delfin/api/v1/storages.py

```diff
@@ -70,12 +70,21 @@ def show(self, req, id):
 
     @wsgi.response(201)
     @validation.schema(schema_storages.create)
-    @coordination.synchronized('storage-create-{body[host]}-{body[port]}')
     def create(self, req, body):
         """Register a new storage device."""
         ctxt = req.environ['delfin.context']
         access_info_dict = body
 
+        # Lock to avoid synchronous creating.
+        if access_info_dict.get('rest_access') is not None:
```

Comment:
What is the dict structure? Any possibility of array being configured with multiple access modes?
(also in future there can be other access modes too) can we optimize it as:
```
for access in constants.ACCESS_TYPE:
     if access_info_dict.get(access):
             host = access_info_dict.get(acess).get('host')
             port = access_info_dict.get(acess).get('port')
             break
```

Reply:
Thanks, have changed.